specs/merge/beacon-chain.md

```diff
@@ -161,16 +170,9 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 
 #### Execution payload processing
 
-##### `get_execution_state`
-
-*Note*: `ExecutionState` class is an abstract class representing Ethereum execution state.
-
-Let `get_execution_state(execution_state_root: Bytes32) -> ExecutionState`  be the function that given the root hash returns a copy of Ethereum execution state. 
-The body of the function is implementation dependent.
-
 ##### `execution_state_transition`
 
-Let `execution_state_transition(execution_state: ExecutionState, execution_payload: ExecutionPayload) -> None` be the transition function of Ethereum execution state. 
+Let `execution_state_transition(execution_state_root: Bytes32, execution_payload: ExecutionPayload, timestamp: uint64) -> None` be the transition function of Ethereum execution state. 
```

Comment:
Also, do we even need to pass in the pre `execution_state_root`? That information is contained within `parent_hash` within the payload and it is assumed that the execution as the post state of the parent (otherwise would throw)

Reply:
Right. Originally, there was an intention to make `execution_state_transition` a pure function of `execution_state` and `execution_payload`. Though, this is not always the case as whether there is an access to the `execution_state` object or there isn't is implementation dependent.

For stateless execution there will be no need in the state root at all as the payload will be self-sufficient in terms of inputs into execution state transition. If we have `timestamp` field in `ExecutionPayload` then it might look like:
```python
verify_execution(execution_payload: ExecutionPayload) -> boolean

...

assert verify_execution(body.execution_payload)
```

Reply:
Right, that's why we were leaning toward that prior structure to begin with. Remove the implicit argument

I still kind of like the prior structure, but don't feel strongly enough to argue for it too much.

You could actually have the same function signature (explicitly passing in state) with stateless as well -- e.g. build a "state" object that is actually a partial and just has the accesses provided from the block. But this isn't really necessary

If we do a stateful verification here, I would argue to remove the explicit state root and make it more like the `verify_execution` function signature above. The extra param isn't used now and wouldn't be used if/when we move to stateless so probably should just not be there

Reply:
Let me apply this change and see how it would look like, we may revert it in case of dislike